projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+const (
+	url        = "http://localhost:8080"
+	maxRetries = 3
+)
+
+func main() {
+	currentTime := time.Now()
+	client := &http.Client{}
+
+	for retries := 0; retries <= maxRetries; retries++ {
+		response, err := handleWeatherRequest(url, client, currentTime, retries)
+		if err != nil {
+			handleError(err)
+			os.Exit(1)
+		}
+		if response != "" {
+			fmt.Fprintln(os.Stdout, response)
+			os.Exit(0)
+		}
+	}
+}
+
+func handleWeatherRequest(url string, client *http.Client, currentTime time.Time, retries int) (string, error) {
+	resp, err := getWeather(url, client)
+	if err != nil {
+		return "", fmt.Errorf("failed to get weather: %v", err)
```

Comment:
Prefer to use `%w` when wrapping errors:

```suggestion
		return "", fmt.Errorf("failed to get weather: %w", err)
```

See https://stackoverflow.com/questions/61283248/format-errors-in-go-s-v-or-w for details